src/vs/workbench/contrib/debug/common/loadedScriptsPicker.ts

```diff
@@ -0,0 +1,105 @@
+
+	const quickPick = quickInputService.createQuickPick<IPickerLoadedScriptItem>();
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "Search loaded scripts by name");
+	quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	const changeListener = quickPick.onDidChangeValue(async () => {
+		quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	});
+	const acceptListener = quickPick.onDidAccept(() => {
+		const selectedItem = quickPick.selectedItems[0];
+		selectedItem.accept();
+		acceptListener.dispose();
+		changeListener.dispose();
+		quickPick.hide();
+		quickPick.dispose();
+	});
+	quickPick.show();
+}
+
+function _getLabelName(source: Source) {
+	const posOfForwardSlash = source.name.lastIndexOf('/');
+	const posOfBackSlash = source.name.lastIndexOf('\\');
+
+	const delimeterPos = posOfForwardSlash > posOfBackSlash ? posOfForwardSlash : posOfBackSlash;
+	const label = source.name.substring(delimeterPos + 1);
+	return label;
+}
+
+async function _getPicksFromSession(session: IDebugSession, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const items: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+	items.push({ type: 'separator', label: session.name });
+	const sources = await session.getLoadedSources();
+
+	sources.forEach((element: Source) => {
+		const pick = _createPick(element, filter, editorService, modelService, languageService);
+		if (pick) {
+			items.push(pick);
+		}
+
+	});
+	return items;
+}
+async function _getPicks(filter: string, sessions: IDebugSession[], editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const loadedScriptPicks: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+
+
+	const picks = await Promise.all(
+		sessions.map((session) => {
+			return _getPicksFromSession(session, filter, editorService, modelService, languageService);
+		})
+	);
+
+	for (const row of picks) {
+		for (const elem of row) {
+			loadedScriptPicks.push(elem);
+		}
+	}
+	return loadedScriptPicks;
+}
+
+function _createPick(source: Source, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): IPickerLoadedScriptItem | undefined {
+
+	const label = _getLabelName(source);
+
+	// manually filter so that headers don't get filtered out
+	const highlights = matchesFuzzy(filter, label, true);
+	if (highlights) {
+		return {
+			label,
+			description: source.name,
```

Comment:
The description should only be the dirname of the source, this duplicates the file name

<img width="199" alt="image" src="https://user-images.githubusercontent.com/323878/175426388-5481c8cc-f1a0-4cb4-9b89-8e5bf5645636.png">